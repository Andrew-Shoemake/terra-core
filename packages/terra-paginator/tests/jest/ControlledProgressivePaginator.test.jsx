import React from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { shallowWithIntl, renderWithIntl } from 'terra-enzyme-intl';
import ControlledProgressivePaginator from '../../src/ControlledProgressivePaginator';

describe('Paginator', () => {
  const defaultRender = renderWithIntl((
    <ControlledProgressivePaginator
      onPageChange={e => typeof e}
      selectedPage={1}
      totalCount={2234}
      itemCountPerPage={20}
    />
  ));

  // Snapshot Tests
  it('should render a Controlled ProgressivePaginator', () => {
    const wrapper = shallowWithIntl(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });
});
