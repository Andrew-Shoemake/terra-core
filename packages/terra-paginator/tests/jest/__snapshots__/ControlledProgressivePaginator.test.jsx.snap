// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator should render a controlled progressive paginator 1`] = `
<ResponsiveElement
  responsiveTo="parent"
  small={
    <div
      aria-label="pagination"
      className="paginator progressive"
      role="navigation"
    >
      <div>
        Page
         
        1
         
        of
         
        112
      </div>
      <div>
        <button
          aria-disabled={true}
          aria-label="first"
          className="nav-link is-disabled"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={null}
          type="button"
        >
          First
        </button>
        <button
          aria-disabled={true}
          aria-label="previous"
          className="nav-link previous is-disabled"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={null}
          type="button"
        >
          <span
            className="icon"
          />
          Previous
        </button>
        <button
          aria-disabled={false}
          aria-label="next"
          className="nav-link next"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex="0"
          type="button"
        >
          Next
          <span
            className="icon"
          />
        </button>
        <button
          aria-disabled={false}
          aria-label="last"
          className="nav-link"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex="0"
          type="button"
        >
          Last
        </button>
      </div>
    </div>
  }
  tiny={
    <div
      aria-label="pagination"
      className="paginator"
      role="navigation"
    >
      <div>
        <button
          aria-disabled={true}
          aria-label="first"
          className="nav-link is-disabled"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={null}
          type="button"
        >
          First
        </button>
        <button
          aria-disabled={true}
          aria-label="previous"
          className="nav-link previous icon-only is-disabled"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={null}
          type="button"
        >
          <span
            className="visually-hidden"
          >
            Previous
          </span>
          <span
            className="icon"
          />
        </button>
      </div>
      <div>
        Page
         
        1
         
        of
         
        112
      </div>
      <div>
        <button
          aria-disabled={false}
          aria-label="next"
          className="nav-link next icon-only"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="visually-hidden"
          >
            Next
          </span>
          <span
            className="icon"
          />
        </button>
        <button
          aria-disabled={false}
          aria-label="last"
          className="nav-link"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex="0"
          type="button"
        >
          Last
        </button>
      </div>
    </div>
  }
/>
`;
